diff --git a/src/create.c b/src/create.c
index c7cf64c..f807a3d 100644
--- a/src/create.c
+++ b/src/create.c
@@ -1746,6 +1746,8 @@ dump_file0 (struct tar_stat_info *st, const char *p,
       if (NAME_FIELD_SIZE - (archive_format == OLDGNU_FORMAT) < size)
 	write_long_link (st);
 
+      xattrs_selinux_get(st, p, -1);
+
       block_ordinal = current_block_ordinal ();
       st->stat.st_size = 0;	/* force 0 size on symlink */
       header = start_header (st);
