diff -urNp tar-1.22-orig/lib/utimens.c tar-1.22/lib/utimens.c
--- tar-1.22-orig/lib/utimens.c	2008-10-30 08:57:49.000000000 +0100
+++ tar-1.22/lib/utimens.c	2010-01-05 20:14:50.348792138 +0100
@@ -120,16 +120,17 @@ gl_futimens (int fd ATTRIBUTE_UNUSED,
     }
 #endif
 #if HAVE_FUTIMENS
-  {
-    int result = futimens (fd, timespec);
+    if (fd >= 0)
+    {
+      int result = futimens (fd, timespec);
 # ifdef __linux__
-    /* Work around the same bug as above.  */
-    if (0 < result)
-      errno = ENOSYS;
+      /* Work around the same bug as above.  */
+      if (0 < result)
+        errno = ENOSYS;
 # endif
-    if (result == 0 || errno != ENOSYS)
-      return result;
-  }
+      if (result == 0 || errno != ENOSYS)
+        return result;
+    }
 #endif
 
   /* The platform lacks an interface to set file timestamps with
diff -urNp tar-1.22-orig/tests/testsuite.dir/60/run tar-1.22/tests/testsuite.dir/60/run
--- tar-1.22-orig/tests/testsuite.dir/60/run	1970-01-01 01:00:00.000000000 +0100
+++ tar-1.22/tests/testsuite.dir/60/run	2010-01-05 20:10:04.867806607 +0100
@@ -0,0 +1,5 @@
+#! /bin/sh
+test "${ZSH_VERSION+set}" = set && alias -g '${1+"$@"}'='"$@"'
+cd '/home/Reset/koji/tar/devel/tar-1.22/tests'
+exec ${CONFIG_SHELL-/bin/sh} "/home/Reset/koji/tar/devel/tar-1.22/tests/./testsuite" -v -d  60 ${1+"$@"}
+exit 1
diff -urNp tar-1.22-orig/tests/testsuite.dir/60/testsuite.log tar-1.22/tests/testsuite.dir/60/testsuite.log
--- tar-1.22-orig/tests/testsuite.dir/60/testsuite.log	1970-01-01 01:00:00.000000000 +0100
+++ tar-1.22/tests/testsuite.dir/60/testsuite.log	2010-01-05 20:10:04.866806051 +0100
@@ -0,0 +1 @@
+60. sparse03.at:21: 60. storing sparse files > 8G (sparse03.at:21): FAILED ()
diff -urNp tar-1.22-orig/tests/testsuite.dir/61/run tar-1.22/tests/testsuite.dir/61/run
--- tar-1.22-orig/tests/testsuite.dir/61/run	1970-01-01 01:00:00.000000000 +0100
+++ tar-1.22/tests/testsuite.dir/61/run	2010-01-05 20:10:04.885806990 +0100
@@ -0,0 +1,5 @@
+#! /bin/sh
+test "${ZSH_VERSION+set}" = set && alias -g '${1+"$@"}'='"$@"'
+cd '/home/Reset/koji/tar/devel/tar-1.22/tests'
+exec ${CONFIG_SHELL-/bin/sh} "/home/Reset/koji/tar/devel/tar-1.22/tests/./testsuite" -v -d  61 ${1+"$@"}
+exit 1
diff -urNp tar-1.22-orig/tests/testsuite.dir/61/testsuite.log tar-1.22/tests/testsuite.dir/61/testsuite.log
--- tar-1.22-orig/tests/testsuite.dir/61/testsuite.log	1970-01-01 01:00:00.000000000 +0100
+++ tar-1.22/tests/testsuite.dir/61/testsuite.log	2010-01-05 20:10:04.885806990 +0100
@@ -0,0 +1 @@
+61. sparse03.at:21: 61. storing sparse files > 8G (sparse03.at:21): FAILED ()
diff -urNp tar-1.22-orig/tests/testsuite.dir/62/run tar-1.22/tests/testsuite.dir/62/run
--- tar-1.22-orig/tests/testsuite.dir/62/run	1970-01-01 01:00:00.000000000 +0100
+++ tar-1.22/tests/testsuite.dir/62/run	2010-01-05 20:10:04.903917213 +0100
@@ -0,0 +1,5 @@
+#! /bin/sh
+test "${ZSH_VERSION+set}" = set && alias -g '${1+"$@"}'='"$@"'
+cd '/home/Reset/koji/tar/devel/tar-1.22/tests'
+exec ${CONFIG_SHELL-/bin/sh} "/home/Reset/koji/tar/devel/tar-1.22/tests/./testsuite" -v -d  62 ${1+"$@"}
+exit 1
diff -urNp tar-1.22-orig/tests/testsuite.dir/62/testsuite.log tar-1.22/tests/testsuite.dir/62/testsuite.log
--- tar-1.22-orig/tests/testsuite.dir/62/testsuite.log	1970-01-01 01:00:00.000000000 +0100
+++ tar-1.22/tests/testsuite.dir/62/testsuite.log	2010-01-05 20:10:04.902917410 +0100
@@ -0,0 +1 @@
+62. sparse03.at:21: 62. storing sparse files > 8G (sparse03.at:21): FAILED ()
diff -urNp tar-1.22-orig/tests/testsuite.dir/63/run tar-1.22/tests/testsuite.dir/63/run
--- tar-1.22-orig/tests/testsuite.dir/63/run	1970-01-01 01:00:00.000000000 +0100
+++ tar-1.22/tests/testsuite.dir/63/run	2010-01-05 20:10:04.921931637 +0100
@@ -0,0 +1,5 @@
+#! /bin/sh
+test "${ZSH_VERSION+set}" = set && alias -g '${1+"$@"}'='"$@"'
+cd '/home/Reset/koji/tar/devel/tar-1.22/tests'
+exec ${CONFIG_SHELL-/bin/sh} "/home/Reset/koji/tar/devel/tar-1.22/tests/./testsuite" -v -d  63 ${1+"$@"}
+exit 1
diff -urNp tar-1.22-orig/tests/testsuite.dir/63/testsuite.log tar-1.22/tests/testsuite.dir/63/testsuite.log
--- tar-1.22-orig/tests/testsuite.dir/63/testsuite.log	1970-01-01 01:00:00.000000000 +0100
+++ tar-1.22/tests/testsuite.dir/63/testsuite.log	2010-01-05 20:10:04.920917687 +0100
@@ -0,0 +1 @@
+63. sparse03.at:21: 63. storing sparse files > 8G (sparse03.at:21): FAILED ()
diff -urNp tar-1.22-orig/tests/testsuite.dir/64/run tar-1.22/tests/testsuite.dir/64/run
--- tar-1.22-orig/tests/testsuite.dir/64/run	1970-01-01 01:00:00.000000000 +0100
+++ tar-1.22/tests/testsuite.dir/64/run	2010-01-05 20:10:04.938927324 +0100
@@ -0,0 +1,5 @@
+#! /bin/sh
+test "${ZSH_VERSION+set}" = set && alias -g '${1+"$@"}'='"$@"'
+cd '/home/Reset/koji/tar/devel/tar-1.22/tests'
+exec ${CONFIG_SHELL-/bin/sh} "/home/Reset/koji/tar/devel/tar-1.22/tests/./testsuite" -v -d  64 ${1+"$@"}
+exit 1
diff -urNp tar-1.22-orig/tests/testsuite.dir/64/testsuite.log tar-1.22/tests/testsuite.dir/64/testsuite.log
--- tar-1.22-orig/tests/testsuite.dir/64/testsuite.log	1970-01-01 01:00:00.000000000 +0100
+++ tar-1.22/tests/testsuite.dir/64/testsuite.log	2010-01-05 20:10:04.937917538 +0100
@@ -0,0 +1 @@
+64. sparse03.at:21: 64. storing sparse files > 8G (sparse03.at:21): FAILED ()
diff -urNp tar-1.22-orig/tests/testsuite.dir/65/run tar-1.22/tests/testsuite.dir/65/run
--- tar-1.22-orig/tests/testsuite.dir/65/run	1970-01-01 01:00:00.000000000 +0100
+++ tar-1.22/tests/testsuite.dir/65/run	2010-01-05 20:10:04.955917347 +0100
@@ -0,0 +1,5 @@
+#! /bin/sh
+test "${ZSH_VERSION+set}" = set && alias -g '${1+"$@"}'='"$@"'
+cd '/home/Reset/koji/tar/devel/tar-1.22/tests'
+exec ${CONFIG_SHELL-/bin/sh} "/home/Reset/koji/tar/devel/tar-1.22/tests/./testsuite" -v -d  65 ${1+"$@"}
+exit 1
diff -urNp tar-1.22-orig/tests/testsuite.dir/65/testsuite.log tar-1.22/tests/testsuite.dir/65/testsuite.log
--- tar-1.22-orig/tests/testsuite.dir/65/testsuite.log	1970-01-01 01:00:00.000000000 +0100
+++ tar-1.22/tests/testsuite.dir/65/testsuite.log	2010-01-05 20:10:04.954917003 +0100
@@ -0,0 +1 @@
+65. sparse03.at:21: 65. storing sparse files > 8G (sparse03.at:21): FAILED ()
diff -urNp tar-1.22-orig/tests/testsuite.dir/66/run tar-1.22/tests/testsuite.dir/66/run
--- tar-1.22-orig/tests/testsuite.dir/66/run	1970-01-01 01:00:00.000000000 +0100
+++ tar-1.22/tests/testsuite.dir/66/run	2010-01-05 20:10:04.973917081 +0100
@@ -0,0 +1,5 @@
+#! /bin/sh
+test "${ZSH_VERSION+set}" = set && alias -g '${1+"$@"}'='"$@"'
+cd '/home/Reset/koji/tar/devel/tar-1.22/tests'
+exec ${CONFIG_SHELL-/bin/sh} "/home/Reset/koji/tar/devel/tar-1.22/tests/./testsuite" -v -d  66 ${1+"$@"}
+exit 1
diff -urNp tar-1.22-orig/tests/testsuite.dir/66/testsuite.log tar-1.22/tests/testsuite.dir/66/testsuite.log
--- tar-1.22-orig/tests/testsuite.dir/66/testsuite.log	1970-01-01 01:00:00.000000000 +0100
+++ tar-1.22/tests/testsuite.dir/66/testsuite.log	2010-01-05 20:10:04.972917019 +0100
@@ -0,0 +1 @@
+66. sparse03.at:21: 66. storing sparse files > 8G (sparse03.at:21): FAILED ()
diff -urNp tar-1.22-orig/tests/testsuite.dir/67/run tar-1.22/tests/testsuite.dir/67/run
--- tar-1.22-orig/tests/testsuite.dir/67/run	1970-01-01 01:00:00.000000000 +0100
+++ tar-1.22/tests/testsuite.dir/67/run	2010-01-05 20:10:04.991924944 +0100
@@ -0,0 +1,5 @@
+#! /bin/sh
+test "${ZSH_VERSION+set}" = set && alias -g '${1+"$@"}'='"$@"'
+cd '/home/Reset/koji/tar/devel/tar-1.22/tests'
+exec ${CONFIG_SHELL-/bin/sh} "/home/Reset/koji/tar/devel/tar-1.22/tests/./testsuite" -v -d  67 ${1+"$@"}
+exit 1
diff -urNp tar-1.22-orig/tests/testsuite.dir/67/testsuite.log tar-1.22/tests/testsuite.dir/67/testsuite.log
--- tar-1.22-orig/tests/testsuite.dir/67/testsuite.log	1970-01-01 01:00:00.000000000 +0100
+++ tar-1.22/tests/testsuite.dir/67/testsuite.log	2010-01-05 20:10:04.990917049 +0100
@@ -0,0 +1 @@
+67. sparse03.at:21: 67. storing sparse files > 8G (sparse03.at:21): FAILED ()
diff -urNp tar-1.22-orig/tests/testsuite.dir/68/run tar-1.22/tests/testsuite.dir/68/run
--- tar-1.22-orig/tests/testsuite.dir/68/run	1970-01-01 01:00:00.000000000 +0100
+++ tar-1.22/tests/testsuite.dir/68/run	2010-01-05 20:10:05.009928244 +0100
@@ -0,0 +1,5 @@
+#! /bin/sh
+test "${ZSH_VERSION+set}" = set && alias -g '${1+"$@"}'='"$@"'
+cd '/home/Reset/koji/tar/devel/tar-1.22/tests'
+exec ${CONFIG_SHELL-/bin/sh} "/home/Reset/koji/tar/devel/tar-1.22/tests/./testsuite" -v -d  68 ${1+"$@"}
+exit 1
diff -urNp tar-1.22-orig/tests/testsuite.dir/68/testsuite.log tar-1.22/tests/testsuite.dir/68/testsuite.log
--- tar-1.22-orig/tests/testsuite.dir/68/testsuite.log	1970-01-01 01:00:00.000000000 +0100
+++ tar-1.22/tests/testsuite.dir/68/testsuite.log	2010-01-05 20:10:05.009928244 +0100
@@ -0,0 +1 @@
+68. sparse03.at:21: 68. storing sparse files > 8G (sparse03.at:21): FAILED ()
diff -urNp tar-1.22-orig/tests/testsuite.dir/69/run tar-1.22/tests/testsuite.dir/69/run
--- tar-1.22-orig/tests/testsuite.dir/69/run	1970-01-01 01:00:00.000000000 +0100
+++ tar-1.22/tests/testsuite.dir/69/run	2010-01-05 20:10:05.027917249 +0100
@@ -0,0 +1,5 @@
+#! /bin/sh
+test "${ZSH_VERSION+set}" = set && alias -g '${1+"$@"}'='"$@"'
+cd '/home/Reset/koji/tar/devel/tar-1.22/tests'
+exec ${CONFIG_SHELL-/bin/sh} "/home/Reset/koji/tar/devel/tar-1.22/tests/./testsuite" -v -d  69 ${1+"$@"}
+exit 1
diff -urNp tar-1.22-orig/tests/testsuite.dir/69/testsuite.log tar-1.22/tests/testsuite.dir/69/testsuite.log
--- tar-1.22-orig/tests/testsuite.dir/69/testsuite.log	1970-01-01 01:00:00.000000000 +0100
+++ tar-1.22/tests/testsuite.dir/69/testsuite.log	2010-01-05 20:10:05.026917108 +0100
@@ -0,0 +1 @@
+69. sparse03.at:21: 69. storing sparse files > 8G (sparse03.at:21): FAILED ()
diff -urNp tar-1.22-orig/tests/testsuite.dir/70/run tar-1.22/tests/testsuite.dir/70/run
--- tar-1.22-orig/tests/testsuite.dir/70/run	1970-01-01 01:00:00.000000000 +0100
+++ tar-1.22/tests/testsuite.dir/70/run	2010-01-05 20:10:05.045930222 +0100
@@ -0,0 +1,5 @@
+#! /bin/sh
+test "${ZSH_VERSION+set}" = set && alias -g '${1+"$@"}'='"$@"'
+cd '/home/Reset/koji/tar/devel/tar-1.22/tests'
+exec ${CONFIG_SHELL-/bin/sh} "/home/Reset/koji/tar/devel/tar-1.22/tests/./testsuite" -v -d  70 ${1+"$@"}
+exit 1
diff -urNp tar-1.22-orig/tests/testsuite.dir/70/testsuite.log tar-1.22/tests/testsuite.dir/70/testsuite.log
--- tar-1.22-orig/tests/testsuite.dir/70/testsuite.log	1970-01-01 01:00:00.000000000 +0100
+++ tar-1.22/tests/testsuite.dir/70/testsuite.log	2010-01-05 20:10:05.044917449 +0100
@@ -0,0 +1 @@
+70. sparse03.at:21: 70. storing sparse files > 8G (sparse03.at:21): FAILED ()
diff -urNp tar-1.22-orig/tests/testsuite.dir/71/run tar-1.22/tests/testsuite.dir/71/run
--- tar-1.22-orig/tests/testsuite.dir/71/run	1970-01-01 01:00:00.000000000 +0100
+++ tar-1.22/tests/testsuite.dir/71/run	2010-01-05 20:10:05.062928453 +0100
@@ -0,0 +1,5 @@
+#! /bin/sh
+test "${ZSH_VERSION+set}" = set && alias -g '${1+"$@"}'='"$@"'
+cd '/home/Reset/koji/tar/devel/tar-1.22/tests'
+exec ${CONFIG_SHELL-/bin/sh} "/home/Reset/koji/tar/devel/tar-1.22/tests/./testsuite" -v -d  71 ${1+"$@"}
+exit 1
diff -urNp tar-1.22-orig/tests/testsuite.dir/71/testsuite.log tar-1.22/tests/testsuite.dir/71/testsuite.log
--- tar-1.22-orig/tests/testsuite.dir/71/testsuite.log	1970-01-01 01:00:00.000000000 +0100
+++ tar-1.22/tests/testsuite.dir/71/testsuite.log	2010-01-05 20:10:05.061916989 +0100
@@ -0,0 +1 @@
+71. sparse03.at:21: 71. storing sparse files > 8G (sparse03.at:21): FAILED ()
diff -urNp tar-1.22-orig/tests/testsuite.dir/72/run tar-1.22/tests/testsuite.dir/72/run
--- tar-1.22-orig/tests/testsuite.dir/72/run	1970-01-01 01:00:00.000000000 +0100
+++ tar-1.22/tests/testsuite.dir/72/run	2010-01-05 20:10:05.080930375 +0100
@@ -0,0 +1,5 @@
+#! /bin/sh
+test "${ZSH_VERSION+set}" = set && alias -g '${1+"$@"}'='"$@"'
+cd '/home/Reset/koji/tar/devel/tar-1.22/tests'
+exec ${CONFIG_SHELL-/bin/sh} "/home/Reset/koji/tar/devel/tar-1.22/tests/./testsuite" -v -d  72 ${1+"$@"}
+exit 1
diff -urNp tar-1.22-orig/tests/testsuite.dir/72/testsuite.log tar-1.22/tests/testsuite.dir/72/testsuite.log
--- tar-1.22-orig/tests/testsuite.dir/72/testsuite.log	1970-01-01 01:00:00.000000000 +0100
+++ tar-1.22/tests/testsuite.dir/72/testsuite.log	2010-01-05 20:10:05.079917866 +0100
@@ -0,0 +1 @@
+72. sparse03.at:21: 72. storing sparse files > 8G (sparse03.at:21): FAILED ()
diff -urNp tar-1.22-orig/tests/testsuite.dir/73/run tar-1.22/tests/testsuite.dir/73/run
--- tar-1.22-orig/tests/testsuite.dir/73/run	1970-01-01 01:00:00.000000000 +0100
+++ tar-1.22/tests/testsuite.dir/73/run	2010-01-05 20:10:05.097917414 +0100
@@ -0,0 +1,5 @@
+#! /bin/sh
+test "${ZSH_VERSION+set}" = set && alias -g '${1+"$@"}'='"$@"'
+cd '/home/Reset/koji/tar/devel/tar-1.22/tests'
+exec ${CONFIG_SHELL-/bin/sh} "/home/Reset/koji/tar/devel/tar-1.22/tests/./testsuite" -v -d  73 ${1+"$@"}
+exit 1
diff -urNp tar-1.22-orig/tests/testsuite.dir/73/testsuite.log tar-1.22/tests/testsuite.dir/73/testsuite.log
--- tar-1.22-orig/tests/testsuite.dir/73/testsuite.log	1970-01-01 01:00:00.000000000 +0100
+++ tar-1.22/tests/testsuite.dir/73/testsuite.log	2010-01-05 20:10:05.096917338 +0100
@@ -0,0 +1 @@
+73. sparse03.at:21: 73. storing sparse files > 8G (sparse03.at:21): FAILED ()
diff -urNp tar-1.22-orig/tests/testsuite.dir/74/run tar-1.22/tests/testsuite.dir/74/run
--- tar-1.22-orig/tests/testsuite.dir/74/run	1970-01-01 01:00:00.000000000 +0100
+++ tar-1.22/tests/testsuite.dir/74/run	2010-01-05 20:10:05.114917494 +0100
@@ -0,0 +1,5 @@
+#! /bin/sh
+test "${ZSH_VERSION+set}" = set && alias -g '${1+"$@"}'='"$@"'
+cd '/home/Reset/koji/tar/devel/tar-1.22/tests'
+exec ${CONFIG_SHELL-/bin/sh} "/home/Reset/koji/tar/devel/tar-1.22/tests/./testsuite" -v -d  74 ${1+"$@"}
+exit 1
diff -urNp tar-1.22-orig/tests/testsuite.dir/74/testsuite.log tar-1.22/tests/testsuite.dir/74/testsuite.log
--- tar-1.22-orig/tests/testsuite.dir/74/testsuite.log	1970-01-01 01:00:00.000000000 +0100
+++ tar-1.22/tests/testsuite.dir/74/testsuite.log	2010-01-05 20:10:05.114917494 +0100
@@ -0,0 +1 @@
+74. sparse03.at:21: 74. storing sparse files > 8G (sparse03.at:21): FAILED ()
diff -urNp tar-1.22-orig/tests/testsuite.dir/75/run tar-1.22/tests/testsuite.dir/75/run
--- tar-1.22-orig/tests/testsuite.dir/75/run	1970-01-01 01:00:00.000000000 +0100
+++ tar-1.22/tests/testsuite.dir/75/run	2010-01-05 20:10:05.132917305 +0100
@@ -0,0 +1,5 @@
+#! /bin/sh
+test "${ZSH_VERSION+set}" = set && alias -g '${1+"$@"}'='"$@"'
+cd '/home/Reset/koji/tar/devel/tar-1.22/tests'
+exec ${CONFIG_SHELL-/bin/sh} "/home/Reset/koji/tar/devel/tar-1.22/tests/./testsuite" -v -d  75 ${1+"$@"}
+exit 1
diff -urNp tar-1.22-orig/tests/testsuite.dir/75/testsuite.log tar-1.22/tests/testsuite.dir/75/testsuite.log
--- tar-1.22-orig/tests/testsuite.dir/75/testsuite.log	1970-01-01 01:00:00.000000000 +0100
+++ tar-1.22/tests/testsuite.dir/75/testsuite.log	2010-01-05 20:10:05.131917628 +0100
@@ -0,0 +1 @@
+75. sparse03.at:21: 75. storing sparse files > 8G (sparse03.at:21): FAILED ()
diff -urNp tar-1.22-orig/tests/testsuite.dir/76/run tar-1.22/tests/testsuite.dir/76/run
--- tar-1.22-orig/tests/testsuite.dir/76/run	1970-01-01 01:00:00.000000000 +0100
+++ tar-1.22/tests/testsuite.dir/76/run	2010-01-05 20:10:05.149930095 +0100
@@ -0,0 +1,5 @@
+#! /bin/sh
+test "${ZSH_VERSION+set}" = set && alias -g '${1+"$@"}'='"$@"'
+cd '/home/Reset/koji/tar/devel/tar-1.22/tests'
+exec ${CONFIG_SHELL-/bin/sh} "/home/Reset/koji/tar/devel/tar-1.22/tests/./testsuite" -v -d  76 ${1+"$@"}
+exit 1
diff -urNp tar-1.22-orig/tests/testsuite.dir/76/testsuite.log tar-1.22/tests/testsuite.dir/76/testsuite.log
--- tar-1.22-orig/tests/testsuite.dir/76/testsuite.log	1970-01-01 01:00:00.000000000 +0100
+++ tar-1.22/tests/testsuite.dir/76/testsuite.log	2010-01-05 20:10:05.148917511 +0100
@@ -0,0 +1 @@
+76. sparse03.at:21: 76. storing sparse files > 8G (sparse03.at:21): FAILED ()
diff -urNp tar-1.22-orig/tests/testsuite.dir/77/run tar-1.22/tests/testsuite.dir/77/run
--- tar-1.22-orig/tests/testsuite.dir/77/run	1970-01-01 01:00:00.000000000 +0100
+++ tar-1.22/tests/testsuite.dir/77/run	2010-01-05 20:10:05.166917570 +0100
@@ -0,0 +1,5 @@
+#! /bin/sh
+test "${ZSH_VERSION+set}" = set && alias -g '${1+"$@"}'='"$@"'
+cd '/home/Reset/koji/tar/devel/tar-1.22/tests'
+exec ${CONFIG_SHELL-/bin/sh} "/home/Reset/koji/tar/devel/tar-1.22/tests/./testsuite" -v -d  77 ${1+"$@"}
+exit 1
diff -urNp tar-1.22-orig/tests/testsuite.dir/77/testsuite.log tar-1.22/tests/testsuite.dir/77/testsuite.log
--- tar-1.22-orig/tests/testsuite.dir/77/testsuite.log	1970-01-01 01:00:00.000000000 +0100
+++ tar-1.22/tests/testsuite.dir/77/testsuite.log	2010-01-05 20:10:05.165917473 +0100
@@ -0,0 +1 @@
+77. sparse03.at:21: 77. storing sparse files > 8G (sparse03.at:21): FAILED ()
