From 1864e189c710414f2f837ddd6b9cf575a05846cd Mon Sep 17 00:00:00 2001
From: Pavel Raiskup <praiskup@redhat.com>
Date: Mon, 13 Jul 2020 08:23:29 +0200
Subject: [PATCH] Bugfix --sparse --diff mode

---
 src/sparse.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/sparse.c b/src/sparse.c
index d642a46..ac78894 100644
--- a/src/sparse.c
+++ b/src/sparse.c
@@ -684,7 +684,7 @@ check_data_region (struct tar_sparse_file *file, size_t i)
 	}
       size_left -= bytes_read;
       mv_size_left (file->stat_info->archive_file_size - file->dumped_size);
-      if (memcmp (blk->buffer, diff_buffer, rdsize))
+      if (memcmp (blk->buffer, diff_buffer, bytes_read))
 	{
 	  report_difference (file->stat_info, _("Contents differ"));
 	  return false;
-- 
2.24.1

